components/Calculator.js

```diff
@@ -0,0 +1,270 @@
+"use client";
+import { formatNumber } from "@/utils/formatNumber";
+import React, { useEffect, useRef, useState } from "react";
+import { useDispatch, useSelector } from "react-redux";
+import { appendNumber } from "../utils/appendNumber";
+import { addOperator } from "../utils/addOperator";
+import { performOperation } from "../utils/performOperation";
+import { clearAll } from "../utils/clearAll";
+import {
+  setHasError,
+  setDisplay,
+  setResult,
+} from "../redux/features/calculatorSlice";
+
+const Calculator = () => {
+  const dispatch = useDispatch();
+  const calculatorState = useSelector((state) => state.calculator);
+
+  const [userLanguage, setUserLanguage] = useState("en-US"); // Default to 'en-US'
+
+  const handleNumberClick = (number) => {
+    appendNumber(
+      number,
+      calculatorState.hasError,
+      calculatorState.blockUserInput,
+      calculatorState.currentOperand,
+      calculatorState.firstOperand,
+      calculatorState.secondOperand,
+      calculatorState.isCommaUsedInFirst,
+      calculatorState.isCommaUsedInSecond,
+      calculatorState.display,
+      dispatch
+    );
+  };
+
+  const handleOperatorClick = (operator) => {
+    addOperator(
+      operator,
+      calculatorState.hasError,
+      calculatorState.firstOperand,
+      calculatorState.secondOperand,
+      calculatorState.isOperatorUsed,
+      calculatorState.display,
+      dispatch
+    );
+  };
+
+  const handleEqualClick = () => {
```

Comment:
Missing or wrongly implemented functionality from requirements "Po wykonaniu operacji można kliknąć kolejny operator, wtedy wynik poprzedniego działania powinien trafić do kolejnego działania na pierwszym miejscu, kliknięty operator ma zostać ustawiony, a druga liczba powinna być pusta."

I think it actually is wrongly implemented. When I do
1*8 and instead of clicking = I do + then it executes as 9 +, it actually should be
8 +
as the previous operation should be executed and + should be set as operator of next operation